packages/eslint-plugin/src/rules/adjacent-overload-signatures.ts

```diff
@@ -1,7 +1,7 @@
 import type { TSESTree } from '@typescript-eslint/utils';
 import { AST_NODE_TYPES } from '@typescript-eslint/utils';
 
-import * as util from '../util';
+import { createRule, getNameFromMember, MemberNameType } from '../util';
```

Comment:
I... kind of like the `import * as util` approach more personally 😅 . When there are a lot of utilities that files swap between, it's nice not having large imports...

Reply:
this was partially a reason why we swapped website-eslint to esbuild and started compiling it from source, as currently we are compiling code co cjs that is hard to safely tree-shake,

changing those imports from * as to named reduces complexity of otput code by removing (not using) some of required helpers, but its still dosn't solve issue with tree-shaking, esp in rollup

see #6716

----------------
https://www.typescriptlang.org/play?target=7&module=1#code/JYWwDg9gTgLgBAKjgQwM5wK42AGzgMyghDgHIA6cgei11IG4AoRgYwgDtV4YBTAD3gBeTNhzl8ECAAoAlIyA
for code:
```ts
import * as util from '../util';

const text = util.foo()
```
currently we are building it to:
```ts
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const util = __importStar(require("../util"));
const text = util.foo();

```

--------------

and for code: https://www.typescriptlang.org/play?target=7&module=1#code/JYWwDg9gTgLgBAbzgMwhOBfFUIjgcgDpCB6AVxmABt8BuAKHoGMIA7AZ3hgFMAPeALwo0ACgCU9IA

```ts
import { foo } from '../util';

const text = foo()
```

we are generating 

```ts
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const util_1 = require("../util");
const text = (0, util_1.foo)();
```